src/App.tsx

```diff
@@ -15,37 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const App: React.FC = () => {
+  const [list, setList] = useState<string[]>(goodsFromServer);
+  const [reverseList, setReverseList] = useState<boolean>(false);
+  const [sortedState, setSortedState] = useState<number>(0);
+
+  const reset = () => {
+    setList(goodsFromServer);
+    setReverseList(false);
+    setSortedState(0);
+  };
+
+  const sortedGoods = (goods: string[]) => {
+    setSortedState(1);
+    const result = [...goods].sort((a, b) => a.localeCompare(b));
+
+    if (reverseList) {
+      return result.reverse();
+    }
+
+    return result;
+  };
+
+  const sortedGoodsLength = (goods: string[]) => {
+    setSortedState(2);
+    if (reverseList) {
+      return [...goods].sort((a, b) => b.length - a.length);
+    }
+
+    return [...goods].sort((a, b) => a.length - b.length);
+  };
+
+  const reversedGoods = (goods: string[]) => {
+    setReverseList(!reverseList);
+
+    return [...goods].reverse();
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortedState === 1 ? '' : 'is-light'}`}
+          onClick={() => setList(sortedGoods(list))}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sortedState === 2 ? '' : 'is-light'}`}
+          onClick={() => setList(sortedGoodsLength(list))}
```

Comment:
Instead of using setList with arrow functions for each button, you could create a Handle Function that makes the test for you and decides which sorting method to use.
Then, pass the handle function here: `onClick={handleSorting}`